libc/string_lenchr.c

```diff
@@ -150,9 +153,41 @@ TEST(string_len, big)
 }
 
 
+TEST_SETUP(string_spn)
+{
+}
+
+
+TEST_TEAR_DOWN(string_spn)
+{
+}
+
+TEST(string_spn, ascii)
```

Comment:
Please think about more cases, for example my first thought is that string s2 which is not present in s1 is not checked (when strspn would return 0). I suppose there won't be much more edge cases, but things like empty strings, bigger buffers, pointing the same string in the same place in memory as you did it in strcspn), pointing to the same strings, which are in different variables and maybe some other things.